src/components/TextInput/TextInput.tsx

```diff
@@ -168,6 +170,7 @@ export const TextInput: FC<TextInputProps> = ({
                 autoComplete={autocomplete ? "on" : "off"}
                 size={size}
                 data-test-id="text-input"
+                spellCheck={spellcheck ?? true}
```

Comment:
[The default](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/spellcheck) seems to be unset so I think we should not enforce `true nor `false` as a default. What do you think?

Reply:
Also, I read that you shouldn't have `<input spellcheck></input>`  if it's undefined so I set the attribute only if we have true or false passed in.